src/bluesearch/database/article.py

```diff
@@ -184,9 +184,16 @@ def abstract(self) -> Generator[str, None, None]:
         str
             The paragraphs of the article abstract.
         """
-        abstract_pars = self.content.findall(".//abstract//p")
+        abstract_pars = self.content.findall("./front/article-meta/abstract//p")
+        children_p = []
         for paragraph in abstract_pars:
-            yield self._element_to_str(paragraph)
+            # Check if paragraph is not child of another paragraph
+            children_p += paragraph.findall(".//p")
+            if paragraph not in children_p:
+                text = self._element_to_str(paragraph)
+                # Check if text is not empty
+                if text:
+                    yield text
```

Comment:
Ending with such logic was what I was talking about when we discussed https://github.com/BlueBrain/Search/issues/449#issuecomment-961230794. Especially, ending with `children_p` to keep a state is exactly the point I was rising.

It feels cumbersome to me to have such a mix of sequential, recursive, and stateful logic. That's why I was advocating for simplifying the method, especially by rethinking the recursivity done by `_element_to_str` and `_inner_text`.

A simplification is to process everything sequentially (`for tag in self.content.find("./front/article-meta/abstract/")`).

Reply:
Sorry I was not aware of this discussion. I can try to experiment and simplify the process.

Reply:
If others are OK with the current implementation, no need to change something. I have actually already approved.

Reply:
The problem is that the paths to all the paragraphs are not consistent (i.e. we can several sections, subsections, ...). So it is not enough to search for `"./front/article-meta/abstract/p"`. Unfortunately, I am not sure we can avoid searching for `//p`, which leads to problem of duplications of paragraphs (exactly as the example code of the issue). 

If you have a better way to handle it, I am totally up to implement it.

Reply:
Like I said below, I think we shouldn't really care what kind of inner nesting the top-level `<p>` have, as it should be covered by `_element_to_str`.

For example in this snippet
```xml
<p>
Some text
<subsection>
	<p>More text</p>
	<p>Even more text [<ref>some ref</ref>]</p>
</subsection>
After subsection
</p>
```
Just applying `_element_to_str` to only the top `<p>` should give all the text and we don't care about the inner `<subsection>` and `<p>` tags.

Reply:
@pafonta @Stannislav What do you think about the new implementation ?

Reply:
The implementation is much cleaner and easy to follow in my opinion!

⚠️  However, there is a regression regarding fixing #496. Indeed, instead of giving
```
('', 'List item 1.')
('', 'List item 2.')
```
now the code is gving
```
('', 'List item 1.\n                List item 2.')
```

Reply:
Do you think it is a problem solving this issue this way ? I have the feeling trying to go to the children paragraphs is going to make things more complicated and we risk to miss the part contained in the parent one. Moreover, I don't think it is a problem to have a bigger paragraphs than several small ones as we mainly we work with sentences. However, I am up to discuss this if you think it is better to keep children paragraphs :)

Reply:
> Do you think it is a problem solving this issue this way ?

Not at the moment.

> However, I am up to discuss this if you think it is better to keep children paragraphs :)

That's OK for now.

Reply:
Perfect! Can I resolve this conversation ?